dlt/load/load.py

```diff
@@ -428,74 +495,51 @@ def load_single_package(self, load_id: str, schema: Schema) -> None:
-        if no_failed_jobs > 0:
-            self.collector.update(
-                "Jobs", no_failed_jobs, message="WARNING: Some of the jobs failed!", label="Failed"
-            )
         # loop until all jobs are processed
+        pending_exception: Optional[Exception] = None
         while True:
             try:
-                remaining_jobs = self.complete_jobs(load_id, jobs, schema)
-                if len(remaining_jobs) == 0:
-                    # get package status
-                    package_jobs = self.load_storage.normalized_packages.get_load_package_jobs(
-                        load_id
-                    )
-                    # possibly raise on failed jobs
-                    if self.config.raise_on_failed_jobs:
-                        if package_jobs["failed_jobs"]:
-                            failed_job = package_jobs["failed_jobs"][0]
-                            raise LoadClientJobFailed(
-                                load_id,
-                                failed_job.job_id(),
-                                self.load_storage.normalized_packages.get_job_failed_message(
-                                    load_id, failed_job
-                                ),
-                            )
-                    # possibly raise on too many retries
-                    if self.config.raise_on_max_retries:
-                        for new_job in package_jobs["new_jobs"]:
-                            r_c = new_job.retry_count
-                            if r_c > 0 and r_c % self.config.raise_on_max_retries == 0:
-                                raise LoadClientJobRetry(
-                                    load_id,
-                                    new_job.job_id(),
-                                    r_c,
-                                    self.config.raise_on_max_retries,
-                                )
+                # we continously spool new jobs and complete finished ones
+                running_jobs, new_pending_exception = self.complete_jobs(
+                    load_id, running_jobs, schema
+                )
+                pending_exception = pending_exception or new_pending_exception
+                # do not spool new jobs if there was a signal
+                if not signals.signal_received() and not pending_exception:
+                    running_jobs += self.start_new_jobs(load_id, schema, running_jobs)
+                self.update_loadpackage_info(load_id)
+
+                if len(running_jobs) == 0:
+                    # if a pending exception was discovered during completion of jobs
+                    # we can raise it now
+                    if pending_exception:
+                        raise pending_exception
                     break
-                # process remaining jobs again
-                jobs = remaining_jobs
                 # this will raise on signal
-                sleep(1)
+                sleep(
```

Comment:
I think you should really create a mock package with 50k jobs. this currently will call list new jobs 10x / second